editors/vscode/client/config.ts

```diff
@@ -4,24 +4,19 @@ import { IDisposable } from './types';
 export class ConfigService implements Config, IDisposable {
   private static readonly _namespace = 'oxc';
   private readonly _disposables: IDisposable[] = [];
-  private _inner: WorkspaceConfiguration;
-  private _runTrigger: Trigger;
-  private _enable: boolean;
-  private _trace: TraceLevel;
-  private _configPath: string;
+  private _inner!: WorkspaceConfiguration;
+  private _runTrigger!: Trigger;
+  private _enable!: boolean;
+  private _trace!: TraceLevel;
+  private _configPath!: string;
```

Comment:
I don't like the non-null assertions here, but I understand the why.  A possible alternative would be to change the class from `implements Config` to contain a property `private _config: Config` and use that instead which wouldn't require non-null assertions.

Previously `ConfigService` was passed to `JSON.stringify` and that was used, but that was replaced with the `toLanguageServerConfig` method.  This is why `implements Config` isn't important anymore.

Reply:
I believe this is a summary of the changes you would need for my suggestion above.
- Replace all the `Config` related properties with a single `_config: Config` property
- Instead of `setSettingsFromWorkspace` modifying the config properties directly, make it `getSettingsFromWorkspace` which returns a `Config`. Update the `_config` property with this in the two places you need it.
- Modify the getters and setters to use the new `_config` property.

Reply:
Added bonus to this is that if we only have a single config property, bugs like the one you are fixing here become less likely.  There won't be two possible ways to retrieve a property, it will always be `_config.<property>` instead of `_inner.<property>` or `_<property>`.